lib/_stream_readable.js

```diff
@@ -490,10 +495,14 @@ Readable.prototype.read = function(n) {
     debug('length less than watermark', doRead);
   }
 
+  // TODO(ronag): Why is this needed?
+  state.errored = state.errored || (this._writableState &&
+    this._writableState.errored);
```

Comment:
@mcollina: I will remove this as soon as I understand why it's needed.

Reply:
I would say we need to sort this before landing this.

Reply:
Can you give me a WIP label. I won't have time to fix this until later this evening.

Reply:
Fixed